src/libs/ajax/Groups.ts

```diff
@@ -0,0 +1,89 @@
+import _ from 'lodash/fp';
+import { authOpts, fetchSam, jsonBody } from 'src/libs/ajax/ajax-common';
+
+export type GroupRole = 'admin' | 'member';
+
+export interface CurrentUserGroupMembership {
+  groupEmail: string;
+  groupName: string;
+  role: GroupRole;
+}
+
+export const Groups = (signal?: AbortSignal) => ({
+  list: async (): Promise<CurrentUserGroupMembership[]> => {
+    const res = await fetchSam('api/groups/v1', _.merge(authOpts(), { signal }));
+    return res.json();
+  },
+
+  group: (groupName: string) => {
+    const root = `api/groups/v1/${groupName}`;
+    const resourceRoot = `api/resources/v1/managed-group/${groupName}`;
+
+    const addRole = (role: GroupRole, email: string): Promise<void> => {
+      return fetchSam(`${root}/${role}/${encodeURIComponent(email)}`, _.merge(authOpts(), { signal, method: 'PUT' }));
+    };
+
+    const removeRole = (role: GroupRole, email: string): Promise<void> => {
+      return fetchSam(
+        `${root}/${role}/${encodeURIComponent(email)}`,
+        _.merge(authOpts(), { signal, method: 'DELETE' })
+      );
+    };
+
+    return {
+      create: (): Promise<void> => {
+        return fetchSam(root, _.merge(authOpts(), { signal, method: 'POST' }));
+      },
+
+      delete: (): Promise<void> => {
+        return fetchSam(root, _.merge(authOpts(), { signal, method: 'DELETE' }));
+      },
+
+      listAdmins: async (): Promise<string[]> => {
+        const res = await fetchSam(`${root}/admin`, _.merge(authOpts(), { signal }));
+        return res.json();
+      },
+
+      listMembers: async (): Promise<string> => {
+        const res = await fetchSam(`${root}/member`, _.merge(authOpts(), { signal }));
+        return res.json();
+      },
```

Comment:
Should this also be returning a `Promise<string[]>`

Reply:
Oops, yes it should.

Reply:
Opened #4412 to fix this.